aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -597,6 +603,65 @@ public TransferObserver upload(String key, File file, ObjectMetadata metadata, C
         return upload(getDefaultBucketOrThrow(), key, file, metadata, cannedAcl, listener);
     }
 
+    /**
+     * Starts uploading the inputStream to the <b>default</b> bucket, using the given key.
+     *
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     * @return A TransferObserver used to track upload progress and state
+     */
+    public TransferObserver upload(String key, InputStream inputStream) throws IOException {
+        // Saves the data as a file in the temporary directory
+        File file = File.createTempFile(UUID.randomUUID().toString(), ".tmp");
```

Comment:
When do these temporary files get removed? Is it possible to remove them as soon as possible after confirming that the upload is successful (this might be a bit tricky)?

Reply:
[File.createTempFile](https://developer.android.com/reference/java/io/File#createTempFile(java.lang.String,%20java.lang.String,%20java.io.File)) automatically appends some UUID or equivalent to ensure the filename is unique, so the prefix doesn't need to be a UUID itself.  Per the docs, it should be at least three characters long, and is recommended that it be a short, meaningful string.  Maybe "awss3"?